### ISSUE TITLE
✨ Making factory functions of `NotEmptyMap.Companion` available on Java

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#525]
## 📝 Description

We would like to make the **experimental** factory functions below for the `NotEmptyMap.Companion` type available for Java users.

```kotlin
fun <K, V> NotEmptyMap.Companion.create(map: Map<K, V>): NotEmptyMap<K, V>
fun <K, V> NotEmptyMap.Companion.createOrNull(map: Map<K, V>): NotEmptyMap<K, V>?

fun <K, V> NotEmptyMap.Companion.of(head: Pair<K, V>, vararg tail: Pair<K, V>): NotEmptyMap<K, V>
```

This means that its compatibility should be tested and its documentation should indicate their usage from Java code.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #354 
- [x] #437 

## ✅ Checklist

- [ ] Make the `create(Map<K, V>)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `createOrNull(Map<K, V>)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `create(Pair<K, V>, vararg Pair<K, V>)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[api binaries]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#checking-the-api-binaries
[test]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#running-tests
[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for stabilizing these functions will be created while resolving #619.
